src/pages/index.js

```diff
@@ -36,42 +38,89 @@ document.addEventListener("DOMContentLoaded", () => {
   const profileEditPopup = new PopupWithForm(
     "#profile-edit-modal",
     (formData) => {
-      userInfo.setUserInfo({
-        name: formData.heading,
-        job: formData.description,
-      });
+      profileEditPopup.renderLoading(true);
+      api
+        .setUserInfo({
+          name: formData.heading,
+          about: formData.description,
+        })
+        .then(() => {
+          userInfo.setUserInfo({
+            name: formData.heading,
+            job: formData.description,
+          });
+          profileEditPopup.close();
+        })
+        .catch((err) => console.error(err))
+        .finally(() => profileEditPopup.renderLoading(false));
     }
   );
 
   const addCardPopup = new PopupWithForm("#add-card-modal", (formData) => {
+    addCardPopup.renderLoading(true);
     const cardData = { name: formData.heading, link: formData.description };
-    const cardElement = getCardElement(cardData);
-    section.addItem(cardElement);
+    api
+      .addCard(cardData)
+      .then((data) => {
+        const cardElement = getCardElement(data);
+        section.addItem(cardElement);
+        addCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => addCardPopup.renderLoading(false));
   });
 
   const imagePopup = new PopupWithImage("#image-modal");
 
-  const confirmPopup = new PopupWithConfirm("#confirm-popup", () => {
-    // Delete the card associated with the confirmation modal
-    cardToRemove.remove(); // Assuming cardToRemove is the card to be deleted
-    confirmPopup.close();
+  const confirmPopup = new PopupWithConfirmation("#confirm-popup");
+  confirmPopup.setEventListeners();
+
+  confirmPopup.setConfirmationHandler(({ cardId, cardElement }) => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardElement.remove();
+        confirmPopup.close();
+      })
+      .catch((err) => console.error(err));
   });
 
   profileEditPopup.setEventListeners();
   addCardPopup.setEventListeners();
   imagePopup.setEventListeners();
-  confirmPopup.setEventListeners();
+
+  const updateAvatarPopup = new PopupWithForm("#profile-popup", (formData) => {
+    updateAvatarPopup.renderLoading(true);
+    api
+      .setUserAvatar({ avatar: formData.description })
+      .then(() => {
+        const profileImageElement = document.querySelector(".profile__image");
+        if (profileImageElement) {
+          profileImageElement.src = formData.description;
```

Comment:
You should use UserInfo class methods for updating the avatar `setAvatar` (implement that method if you don't have it)